### DISCUSSION TITLE
Choose program directory selector or even run an Installer

### DISCUSSION THREAD

**Discussion by osc2nuke** [NativePHP/laravel#182] [Ideas]
First i explain what i am making, so you can visualize the reason and purpose of this idea, after that i believe you are able to imagine how handy it can be for other NativePHP Desktop Apps.

I am building an IPTV player using **FREE AND PUBLICLY AVAILABLE** tv channel streams via:
https://github.com/iptv-org/iptv

What it currently does is that i upload a playlist into the App:
`https://iptv-org.github.io/iptv/categories/animation.m3u`
As the above example playlist only has a few channels, the following one has over 10.000 FREE tv channels from all over the world.
`https://iptv-org.github.io/iptv/index.m3u`

After i download the file, i upload that playlist to my APP, read that file and convert it to a CSV.
As Laravels `DB::` functionality can insert data to the nativephp.sqlite database only one row at a time - what is to much time consuming for a desktop APP- i use the PHP build-in `exec()` or `shell_exec()` commands to insert the CSV file via sqlite.
But of course not everyone has sqlite3 installed on their machine.

The same scenario occurs when i want the client give the option to play the TV streams with their desired already installed MEDIA PLAYER(s) on their machine.
So while i am developing the APP i have to hardcode the path to - in my case to test - VLC Player.

You can probably already imagine that this ends up in catastrophic coding when your APP depends on settings.

**Now here comes my idea!**
What if there would be some kind of installer UI what starts on first run where client can fill out options or leave as *default configuration.
[ * _You not always now this when your APP can be used on Win, Lin ore Mac os_]

Later inside your own personal built APP can have a settings page to change by desired program or other functionalities .
For example to open your default: text-editor, browser, image editing software, Etc..

So ... what do you think, should 'something' like that be built-in or has it to be a separate package?

I am curious what @mpociot generally thinks about it!

**Comment by osc2nuke** [comment]
FYI, the idea occured because i dont know (yet) how to make a selector that picks only the program location from the client, as i only worked with file-upload forms before.
So i am not aware of how-to-do ( when that feature already exist) or must use trickery.
I also got a little inspired to start above discussion because of user @MasterRO94 
what found a solution on his own question:
https://github.com/NativePHP/laravel/discussions/173
As for my personal use i then might get away with:
https://github.com/anlutro/laravel-settings

**Comment by pb30** [comment]
I would say you don't need an "installer" in the typical sense, but just a different first-run experience. Have a simple [JSON settings file in `appData`](https://nativephp.com/docs/1/digging-deeper/files), or a SQLite database if you need something more. If it doesn't contain the paths you need, show a "welcome" UI with options to select the paths using [Dialogs](https://nativephp.com/docs/1/the-basics/dialogs).

**Reply by osc2nuke** [reply]
Yes yes, i agree, i concluded the same after reading back my idea, but i not edited it, as giving the impression of an "installer" still counts.
I was not aware of the Dialogs options, thank you for pointing that out!
